src/mixer/state.ts

```diff
@@ -998,74 +1175,129 @@ export const mixerMiddleware: Middleware<{}, RootState, Dispatch<any>> = (
+      break;
+  }
+};
+
 export const mixerKeyboardShortcutsMiddleware: Middleware<
   {},
   RootState,
   Dispatch<any>
 > = (store) => {
+  if (process.env.REACT_APP_BAPSICLE_INTERFACE) {
+    // The F keys will only work in places like Electron (NeutronStudio) where they don't trigger browser functions.
+    Keys("f1", () => {
+      store.dispatch(handleKeyboardShortcut(store, 0, "PLAY"));
+    });
+    Keys("f2", () => {
+      store.dispatch(handleKeyboardShortcut(store, 0, "PAUSE"));
+    });
+    Keys("f3", () => {
+      store.dispatch(handleKeyboardShortcut(store, 0, "STOP"));
+    });
+    Keys("f5", () => {
+      store.dispatch(handleKeyboardShortcut(store, 1, "PLAY"));
+    });
+    Keys("f6", () => {
+      store.dispatch(handleKeyboardShortcut(store, 1, "PAUSE"));
+    });
+    Keys("f7", () => {
+      store.dispatch(handleKeyboardShortcut(store, 1, "STOP"));
+    });
+    Keys("f9", () => {
+      store.dispatch(handleKeyboardShortcut(store, 2, "PLAY"));
+    });
+    Keys("f10", () => {
+      store.dispatch(handleKeyboardShortcut(store, 2, "PAUSE"));
+    });
+    Keys("f11", () => {
+      store.dispatch(handleKeyboardShortcut(store, 2, "STOP"));
+    });
+  }
   Keys("q", () => {
-    store.dispatch(play(0));
+    store.dispatch(handleKeyboardShortcut(store, 0, "PLAY"));
   });
   Keys("w", () => {
-    store.dispatch(pause(0));
+    store.dispatch(handleKeyboardShortcut(store, 0, "PAUSE"));
   });
   Keys("e", () => {
-    store.dispatch(stop(0));
+    store.dispatch(handleKeyboardShortcut(store, 0, "STOP"));
   });
   Keys("r", () => {
-    store.dispatch(play(1));
+    store.dispatch(handleKeyboardShortcut(store, 1, "PLAY"));
   });
   Keys("t", () => {
-    store.dispatch(pause(1));
+    store.dispatch(handleKeyboardShortcut(store, 1, "PAUSE"));
   });
   Keys("y", () => {
-    store.dispatch(stop(1));
+    store.dispatch(handleKeyboardShortcut(store, 1, "STOP"));
   });
   Keys("u", () => {
-    store.dispatch(play(2));
+    store.dispatch(handleKeyboardShortcut(store, 2, "PLAY"));
   });
   Keys("i", () => {
-    store.dispatch(pause(2));
+    store.dispatch(handleKeyboardShortcut(store, 2, "PAUSE"));
   });
   Keys("o", () => {
-    store.dispatch(stop(2));
+    store.dispatch(handleKeyboardShortcut(store, 2, "STOP"));
   });
 
-  Keys("a", () => {
-    store.dispatch(setVolume(0, "off"));
-  });
-  Keys("s", () => {
-    store.dispatch(setVolume(0, "bed"));
-  });
-  Keys("d", () => {
-    store.dispatch(setVolume(0, "full"));
-  });
-  Keys("f", () => {
-    store.dispatch(setVolume(1, "off"));
-  });
-  Keys("g", () => {
-    store.dispatch(setVolume(1, "bed"));
-  });
-  Keys("h", () => {
-    store.dispatch(setVolume(1, "full"));
-  });
-  Keys("j", () => {
-    store.dispatch(setVolume(2, "off"));
-  });
-  Keys("k", () => {
-    store.dispatch(setVolume(2, "bed"));
-  });
-  Keys("l", () => {
-    store.dispatch(setVolume(2, "full"));
-  });
+  if (!process.env.REACT_APP_BAPSICLE_INTERFACE) {
```

Comment:
Might it be worth keeping these around even in bapsicle mode? It'd make switching between webstudio and baps easier, as people don't have to relearn their muscle memory. Feel free to overrule me though.

Reply:
hmm, it was to mainly get rid of the "i'm typing into the search box" bug where possible. Especially given at least the studio keyboards are labeled with BAPS keys.

Reply:
Hrm, I see what you're getting at, but I can also imagine a hypothetical future scenario where you have a bapsicle backend but browser frontend, where that won't work.

Eh, we can cross that bridge when we come to it.